src/main.rs

```diff
@@ -65,7 +64,18 @@ impl Args {
 }
 
 fn main() -> CliResult {
-    let cli = CliArgs::parse();
+    let cli = match CliArgs::try_parse() {
+        Ok(cli) => cli,
+        Err(error) => {
+            if matches!(
+                error.kind,
+                clap::ErrorKind::UnknownArgument | clap::ErrorKind::InvalidSubcommand
+            ) {
+                return try_external_subcommand_execution();
+            }
+            return Err(color_eyre::eyre::eyre!(error));
```

Comment:
This feels a bit funky to have it fail and then go into trying to run an external command, since we'd potentially run into the problem of not giving the right error messages if a user tries to input an invalid command but not a custom command.

Not sure what's the best way of avoiding that situation, but maybe an extra flag/subcommand like `--custom` to tell clap that we're trying to go into our own dynamic subcommand?

Reply:
Now the error is `"unknown command" command or near-cli-"unknown command" extension does not exist`.
Clap allows us to tune it in the future. There is a list of errors that we can handle differently:
https://docs.rs/clap/2.33.0/clap/enum.ErrorKind.html